src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1174,6 +1174,27 @@ class DateGenerator : public Generator<bsoncxx::types::b_date> {
     const UniqueGenerator<int64_t> _date;
 };
 
+/** `{^BinData: {numBytes: 32}}` */
+class BinDataGenerator : public Generator<bsoncxx::types::b_binary> {
+public:
+    BinDataGenerator(const Node& node, GeneratorArgs generatorArgs)
+        : _node{node} {}
+
+    bsoncxx::types::b_binary evaluate() override {
+        int64_t numBytes = _node["numBytes"].maybe<int64_t>().value_or(32);
+        uint8_t bytesArr[numBytes];
+        for (int i = 0; i < numBytes; i++) {
+            bytesArr[i] = rand();
+        }
```

Comment:
Alternatively I can just hardcode a byte array like 
```
constexpr uint8_t localMasterKey[] = {
            0x77, 0x1f, 0x2d, 0x7d, 0x76, 0x74, 0x39, 0x08, 0x50, 0x0b, 0x61, 0x14, 0x3a, 0x07};
```
and return a Binary built from that, if that's better!

Reply:
Hmm maybe have the generator always return the same value, since that's more what we want in our use-case, but you can generate that single value by calling a helper function 'genRandBinData()' which is basically your current 'evaluate()' method? That way someone could more easily extend/parameterize this to generate randomness in the future if they wanted.

What do you think?

Reply:
Do you mean move what I currently have in evaluate to a `genRandBinData()` and separately have the generator return the same value? Or just have the `genRandBinData()` return a hardcoded value for now? I just worry that the name would be confusing if I am not actually returning a random BinData

Reply:
Yes the first one! The generator's constructor can maybe call the random one once and store the result?